### PR TITLE
Added memoised parser

### PR DESCRIPTION
**Pull request by zesterer**

### CODE REVIEW

**Comment by zesterer** [comment]
Closes #74

**Comment by Zij-IT** [comment]
And #84 :)

File: Cargo.toml [hunk 1]
```diff
@@ -14,13 +14,15 @@ exclude = [
...
-default = ["std", "spill-stack"]
+default = ["std", "spill-stack", "memoisation"]
 # Integrate with the standard library
 std = []
 # Enable nightly-only features like better compiler diagnostics
 nightly = []
 # Allows deeper recursion by dynamically spilling stack state on to the heap
 spill-stack = ["stacker", "std"]
+# Allows parser memoisation, speeding up heavily back-tracking parsers and allowing left recursion
+memoisation = []
```
Comment (zesterer) [inline Cargo.toml:25]: Maybe we should use the US spelling? Or a different feature name?

Comment (CraftSpider) [reply Cargo.toml:25]: I'm trying to think of a different name, and can't think of one. I personally don't feel strongly about the spelling, but I wish we could maybe just have two features with the different spellings that require each-other :P

**Pull request merged**